### DISCUSSION TITLE
Target offset based on direction of travel

### DISCUSSION THREAD

**Discussion by spirosgkogkas** [ramokz/phantom-camera#217] [Question / Help Wanted] [answered]
Hello,
First of all thx for this awesome plugin, I would like to know is there any built-in way to have a look-ahead kind of movement with the target offset?

**Comment by ramokz** [comment, marked as answer]
There isn't at the minute, no.

A PR (#160) was shared a little while back, but I am looking for it to be changed a bit so it's a bit more dynamic based on velocity rather than a fixed offset value.

No too sure when it'll be added, but it's planned!

**Reply by spirosgkogkas** [reply]
Thank you for answering, I would really like to see that implemented because the plugin is really helpful.
I tried to merge the commit with the lookahead however the camera transitions too fast.

https://github.com/ramokz/phantom-camera/assets/69967865/250ad2d7-5829-402f-b093-f357f4809d8a

Any suggestions?
By the way even any resources to learn how to implement those stuff would be really helpful, maybe I could contribute a bit since I am new to the game development.

Thanks in advance!

**Reply by ramokz** [reply]
So I mention [here](https://github.com/ramokz/phantom-camera/pull/160#issuecomment-1872613344) what the intended behaviour ought to be. Basically for the same reason you mention is that the look-ahead value is too sudden, which can lead to very jarring results.
